packages/taco/src/taco-config-validator.ts

```diff
@@ -0,0 +1,339 @@
+}
+
+/**
+ * TACo Configuration Validator
+ *
+ * Validates TACo client configurations, domains, and provider compatibility.
+ * Provides both fast and full validation methods for TACo operations.
+ */
+export class TacoConfigValidator {
+  /**
+   * Get all supported TACo domain names
+   * @returns {DomainName[]} Array of supported TACo domain names ('lynx', 'tapir', 'mainnet')
+   */
+  static getSupportedDomains(): DomainName[] {
+    return Object.values(DOMAINS).map((domain) => domain.domain);
+  }
+
+  /**
+   * Check if domain is valid
+   * @param domain - TACo domain name to check ('lynx', 'tapir', 'mainnet')
+   * @returns {boolean} True if domain exists
+   */
+  static isValidDomain(domain: DomainName): boolean {
+    return !!domain && this.getSupportedDomains().includes(domain);
+  }
+
+  /**
+   * Get expected chain ID for domain from DOMAINS configuration
+   * @param domain - Domain name to look up
+   * @returns {number | undefined} Chain ID for the domain, undefined if not found
+   * @private
+   */
+  private static getExpectedChainId(domain: string): number | undefined {
+    const domainEntry = Object.values(DOMAINS).find(
+      (domainConfig) => domainConfig.domain === domain,
+    );
+    return domainEntry?.chainId;
+  }
+
+  /**
+   * Validate ritual ID (basic validation - positive number only)
+   * @param domain - Domain name (unused but kept for API compatibility)
+   * @param ritualId - Ritual ID to validate
+   * @returns {boolean} True if valid (positive number)
+   */
+  static isValidRitualId(domain: DomainName, ritualId: number): boolean {
+    return typeof ritualId === 'number' && ritualId > 0;
+  }
+
+  /**
+   * Validate provider compatibility with domain
+   * @param domain - Domain name
+   * @param provider - Provider to validate (viem PublicClient or ethers Provider)
+   * @returns {Promise<boolean>} True if provider is valid for domain
+   */
+  static async isValidProvider(
+    domain: DomainName,
+    provider: PublicClient | ethers.providers.Provider,
+  ): Promise<boolean> {
+    let chainId: number;
+
+    try {
+      // Try to detect provider type and get chain ID safely
+      if (provider && typeof provider === 'object') {
+        // Check if it's a viem PublicClient (has getChainId method)
+        if (
+          'getChainId' in provider &&
+          typeof provider.getChainId === 'function'
+        ) {
+          chainId = await (provider as PublicClient).getChainId();
+        }
+        // Check if it's an ethers Provider (has getNetwork method)
+        else if (
+          'getNetwork' in provider &&
+          typeof provider.getNetwork === 'function'
+        ) {
+          const network = await (
+            provider as ethers.providers.Provider
+          ).getNetwork();
+          chainId = network.chainId;
+        } else {
+          // Unknown provider type
+          return false;
+        }
+      } else {
+        // Invalid provider
+        return false;
+      }
```

Comment:
This logic seems overly complex...anything we can do to simplify this?

Reply:
I extracted a condition check above and I hope it is somehow simple enough now.